includes/admin/meta-boxes/views/html-product-attribute.php

```diff
@@ -14,24 +14,27 @@
 			<tbody>
 				<tr>
 					<td class="attribute_name">
-						<label><?php _e( 'Name', 'woocommerce' ); ?>:</label>
+						<label><?php esc_html_e( 'Name', 'woocommerce' ); ?>:</label>
 
 						<?php if ( $attribute->is_taxonomy() ) : ?>
 							<strong><?php echo esc_html( wc_attribute_label( $attribute->get_name() ) ); ?></strong>
-							<input type="hidden" name="attribute_names[<?php echo $i; ?>]" value="<?php echo esc_attr( $attribute->get_name() ); ?>" />
+							<input type="hidden" name="attribute_names[<?php echo esc_attr( $i ); ?>]" value="<?php echo esc_attr( $attribute->get_name() ); ?>" />
 						<?php else : ?>
-							<input type="text" class="attribute_name" name="attribute_names[<?php echo $i; ?>]" value="<?php echo esc_attr( $attribute->get_name() ); ?>" />
+							<input type="text" class="attribute_name" name="attribute_names[<?php echo esc_attr( $i ); ?>]" value="<?php echo esc_attr( $attribute->get_name() ); ?>" />
 						<?php endif; ?>
 
-						<input type="hidden" name="attribute_position[<?php echo $i; ?>]" class="attribute_position" value="<?php echo esc_attr( $attribute->get_position() ); ?>" />
+						<input type="hidden" name="attribute_position[<?php echo esc_attr( $i ); ?>]" class="attribute_position" value="<?php echo esc_attr( $attribute->get_position() ); ?>" />
 					</td>
 					<td rowspan="3">
-						<label><?php _e( 'Value(s)', 'woocommerce' ); ?>:</label>
+						<label><?php esc_html_e( 'Value(s)', 'woocommerce' ); ?>:</label>
 
-						<?php if ( $attribute->is_taxonomy() && ( $attribute_taxonomy = $attribute->get_taxonomy_object() ) ) : ?>
+						<?php
+						$attribute_taxonomy = $attribute->get_taxonomy_object();
```

Comment:
This should be left as it was or tweaked; you're getting a taxonomy object before checking if it's a taxonomy.